.buildkite/pipeline.yml

```diff
@@ -46,3 +46,28 @@ steps:
                   env:
                     - BUILDKITE_PULL_REQUEST=${BUILDKITE_PULL_REQUEST}
                     - GITHUB_PR_LABELS=${GITHUB_PR_LABELS}
+
+  - label: "Trigger x-pack/Auditbeat"
+    plugins:
+      - monorepo-diff#v1.0.1:
+          diff: "git diff --name-only HEAD~1"
```

Comment:
```suggestion
          diff: "git diff --name-only ${GITHUB_PR_TARGET_BRANCH}...HEAD"
```

Reply:
This change will break pipelines that run on main and release branches. I propose to create a script that handles both cases: merge commit and PR.

Reply:
As discussed in yesterday's meeting, once everything is done properly here, we will look at the `main` changes needed in the later phase(https://github.com/elastic/ingest-dev/issues/3072).

Reply:
That's right. Then why should we change it now?

Reply:
because the `HEAD~1` in the current context is not doing what we want it to do, but `${GITHUB_PR_TARGET_BRANCH}...HEAD` does.

Reply:
> That's right. Then why should we change it now?

We need to change it because we can not iterate on our PRs without it; as we keep pushing commits in an already open PR we aren't testing the latest commit.

Reply:
Should I update the previously merged pipelines like [Auditbeat](https://github.com/elastic/beats/blob/main/.buildkite/pipeline.yml#L7) and [Heartbeat](https://github.com/elastic/beats/blob/main/.buildkite/pipeline.yml#L30) within this PR?

Reply:
> Should I update the previously merged pipelines like [Auditbeat](https://github.com/elastic/beats/blob/main/.buildkite/pipeline.yml#L7) and [Heartbeat](https://github.com/elastic/beats/blob/main/.buildkite/pipeline.yml#L30) within this PR?

Separately please for the sake of readability/reviewability.

Reply:
We are tracking the improvements needed here in https://github.com/elastic/ingest-dev/issues/3159